File: 프로그래머스/unrated/181883. 수열과 구간 쿼리 1/수열과 구간 쿼리 1-1.java
import java.util.Arrays;
import java.util.stream.IntStream;

class Solution {
	public int[] solution( int[] arr, int[][] queries ) {
		Arrays.stream( queries )
			  .flatMapToInt(
				  query -> IntStream.rangeClosed( query[ 0 ], query[ 1 ] ) ) // query[0] <= i <=query[1] 에 부합하는 Int 반환
			  .forEach( i -> arr[ i ] += 1 );
		return arr;
	}
}

File: 프로그래머스/unrated/181883. 수열과 구간 쿼리 1/수열과 구간 쿼리 1-1.js
function solution(arr, queries) {
  queries.map(([s, e]) => {
    Array.from(
      { length: e - s + 1 }, // 길이가 s <= 길이 <= e 에 부합하는 배열 생성
      (_, idx) => (arr[s + idx] += 1) // 생성된 배열의 인덱스를 s 로부터 더하면서 arr[s] 부터 arr[e] 까지 +1을 해준다.
    );
  });
  return arr;
}